geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -0,0 +1,155 @@
+
+    val response: GeoMesaCoprocessorResponse = try {
+
+      scanList.foreach(scan => {
+        scan.setFilter(filterList)
+        // TODO: Explore use of MultiRangeFilter
+        val scanner = env.getRegion.getScanner(scan)
+
+        try {
+          val results = new java.util.ArrayList[Cell]
+
+          def processResults() = {
+            for (cell <- results) {
+              // TODO: GEOMESA-1868 Set the Feature ID in the GeoMesa Coprocessor
+              val sf = serializer.deserialize(cell.getValueArray, cell.getValueOffset, cell.getValueLength)
+              aggregator.aggregate(sf)
+            }
+            results.clear()
+          }
+
+          @tailrec
+          def readScanner(): Unit = {
+            if(scanner.next(results)) {
+              processResults()
+              readScanner()
+            } else {
+              processResults()
+            }
+          }
+
+          readScanner()
+        } finally {
+          CloseQuietly(scanner)
+        }
+      })
+
+      val result: Array[Byte] = aggregator.encodeResult()
+      GeoMesaCoprocessorResponse.newBuilder.setSf(ByteString.copyFrom(result)).build
+    } catch {
+      case ioe: IOException =>
+        ResponseConverter.setControllerException(controller, ioe)
+        null
+      case cnfe: ClassNotFoundException =>
+        throw cnfe
+      case dse: DeserializationException =>
+        throw dse
+    }
+
+    done.run(response)
+  }
+}
+
+object GeoMesaCoprocessor {
+  /**
+    * It gives the combined result of pairs received from different region servers value of a column for a given column family for the
+    * given range. In case qualifier is null, a min of all values for the given
+    * family is returned.
+    *
+    * @param table
+    * @return HashMap result;
+    * @throws Throwable
+    */
+  def execute(table: Table, options: Array[Byte]): List[ByteString] = {
+    val requestArg: GeoMesaCoprocessorRequest = GeoMesaCoprocessorRequest.newBuilder().setOptions(ByteString.copyFrom(options)).build()
+
+    val callBack: GeoMesaHBaseCallBack = new GeoMesaHBaseCallBack()
+
+    table.coprocessorService(classOf[GeoMesaCoprocessorService], null, null, new Call[GeoMesaCoprocessorService, ByteString]() {
+      override def call(instance: GeoMesaCoprocessorService): ByteString = {
+        val controller: RpcController = new GeoMesaHBaseRpcController()
+        val rpcCallback: BlockingRpcCallback[GeoMesaCoprocessorResponse] =
+          new BlockingRpcCallback[GeoMesaCoprocessorResponse]()
+        instance.getResult(controller, requestArg, rpcCallback)
+        val response: GeoMesaCoprocessorResponse = rpcCallback.get
+        if (controller.failed()) {
+          throw new IOException(controller.errorText())
+        }
+        response.getSf
+      }
+    },
+      callBack
+    )
+    callBack.getResult
+  }
+
+  def getAggregator(options: Map[String, String]): GeoMesaHBaseAggregator = {
```

Comment:
I don't think so.  I'll mark this as package private to indicate scope more clearly.